.github/actions/kots-e2e/action.yml

```diff
@@ -58,23 +64,42 @@ inputs:
   gke-service-key:
     description: 'Service key to authenticate a Google Cloud service account'
     required: false
+  replicated-api-token:
+    description: 'Replicated API token for C11y matrix'
+    required: false
+
+outputs:
+  cluster-id:
+    description: 'ID of the c11y matrix cluster created by create-cluster action'
+    value: ${{ steps.create-cluster.outputs.cluster-id }}
 
 runs:
   using: "composite"
   steps:
-    - name: Generate GKE cluster name
-      if: inputs.platform == 'gke'
+    - name: Create a GKE cluster
+      if: inputs.k8s-distribution == 'gke'
       uses: ./.github/actions/kots-gke-create
       with:
         service-key: ${{ inputs.gke-service-key }}
 
-    # TODO: use k3d eventually but this is difficult to support with ${{ inputs.k8s-version }}
-    - name: Generate a k3s cluster
-      if: inputs.platform == 'k3s'
+    - name: Create a local k3s cluster
+      if: inputs.k8s-distribution == 'k3s-local'
       uses: replicatedhq/action-k3s@main
       with:
         version: ${{ inputs.k8s-version }}
-        arguments: ${{ inputs.k8s-arguments }}
+
+    - name: Create a c11y matrix cluster 
+      id: create-cluster 
+      uses: replicatedhq/replicated-actions/create-cluster@v1
+      if: ${{ inputs.k8s-distribution == 'k3s' || inputs.k8s-distribution == 'kind' || inputs.k8s-distribution == 'eks' }}
+      with: 
+        api-token: ${{ inputs.replicated-api-token }}
+        kubernetes-distribution: ${{ inputs.k8s-distribution }}
+        kubernetes-version: ${{ inputs.k8s-version }}
+        cluster-name: ${{ github.ref_name }}-${{ inputs.k8s-distribution }}-${{ inputs.k8s-version }}
```

Comment:
how about using the run id instead of ref_name? ref_name is the branch name IIRC, and it would be great to differentiate between runs.